### ISSUE TITLE
Stop using globalThis

### ISSUE THREAD

**Issue by iowxy** [reduxjs/react-redux#2053]
### What version of React, ReactDOM/React Native, Redux, and React Redux are you using?

- React Redux: 8.1.1

### What is the current behavior?

```
globalThis is not defined
```

https://github.com/foca-js/foca/issues/40
https://github.com/NervJS/taro/issues/14082

According the documentation, keyword `globalThis` is disabled on alipay mini-program and polyfill is helpless, could you please replace it?

### What is the expected behavior?

Replace `globalThis`

### Which browser and OS are affected by this issue?

Chinese alipay mini-program

### Did this work in previous versions of React Redux?

- [X] Yes

**Comment by iowxy** [comment]
The documentation is here: https://opendocs.alipay.com/mini/framework/implementation-detail

**Comment by phryneas** [comment]
`globalThis` is part of the ECMAScript standard since ECMAScript 2020 - and has been implemented in most runtimes well before that.

We are at an impasse here, if we want to access a global value: there are runtimes that only have `window` and other runtimes that have `global` or `self` - usually not all of them. `globalThis` has been added to address that.

It is sad to see that there are still runtimes out there that do not adhere to the standard - or in this case, if my chinese translator is correct, outright disable this feature on purpose.

If I read that correctly, they deliberately disable any kind of global access, which would explain why the polyfill is not working for you - but that also means that we could not fall back to `window`, `global` or `self`, since they also disabled all of those.  
If I am honest, this is a ridiculous choice on their side. Is there any way you can define a global variable in that runtime?

**Comment by phryneas** [comment]
I guess we can in that case fall back to module scope, essentially going back to pre-8.1 behavior. See b37e57e9eb8981b03b92a09c58bd15664cb64a12 .

Could you please try if this build solves your problem and still works without anything breaking?
```
npm i https://pkg.csb.dev/reduxjs/react-redux/commit/b37e57e9/react-redux
```

**Comment by phryneas** [comment]
This should be solved in [v8.1.2](https://github.com/reduxjs/react-redux/releases/tag/v8.1.2).
